morseMessenger/flashHook.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react';
+import { morseDictionary, DictOptions } from './morseDictionary';
+
+export default function useFlash() {
+  const TORCH_ON = 'torch';
+  const TORCH_OFF = 'off';
+  const PAUSE = 1000;
+  const FLASH_PAUSE = 400;
+  const FLASH_SHORT = 100;
+  const FLASH_LONG = 700;
+  const [flashMode, setFlashMode] = useState(TORCH_OFF);
+  const [buttonDisabled, setButtonDisabled] = useState(false);
+
+  function torchOffAfter(timeInMs: number) {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        setFlashMode(TORCH_OFF);
+        resolve();
+      }, timeInMs);
+    });
+  }
+  function torchOnAfter(timeInMs: number) {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        setFlashMode(TORCH_ON);
+        resolve();
+      }, timeInMs);
+    });
+  }
+  function short(): Promise<any> {
+    return new Promise((resolve, reject) => {
+      torchOnAfter(FLASH_PAUSE)
+        .then(() => torchOffAfter(FLASH_SHORT))
+        .then(resolve);
+    });
+  }
+
+  function long() {
+    return new Promise((resolve, reject) => {
+      torchOnAfter(FLASH_PAUSE)
+        .then(() => torchOffAfter(FLASH_LONG))
+        .then(resolve);
+    });
+  }
+
+  function pause() {
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve();
+      }, PAUSE);
+    });
+  }
+
+  function translateToMorse(sentence: string) {
+    const letters = sentence.toLowerCase().split('');
+    const morseCode: Array<string> = [];
+    letters.forEach((letter: string) =>
+      morseCode.push(...morseDictionary[letter], DictOptions.pause)
+    );
+    return morseCode;
+  }
+
+  async function startMorseForInput(sentence: string): Promise<void> {
+    setButtonDisabled(true);
+    const morseCode = translateToMorse(sentence);
+    for (let i = 0; i < morseCode.length; i++) {
+      if (morseCode[i] === DictOptions.s) {
+        await short();
+      } else if (morseCode[i] === DictOptions.l) {
+        await long();
+      } else if (morseCode[i] === DictOptions.pause) {
+        await pause();
+      }
+    }
+    setButtonDisabled(false);
```

Comment:
I don't think we should keep `setButtonDisabled` here. This hook/function shouldn't know that you will use it in a button.

It would be better to add such logic e.g. in the button itself:
```ts
const { disableOnPress, onPress } = props; // boolean prop
const [disabled, setDisabled] = useState(false);
const onButtonPress = useCallback(async () => {
  disableOnPress && setDisabled(true);
  try {
    await onPress();
  } finally {
    disableOnPress && setDisabled(false);
  }
}, [onPress, disableOnPress, setDisabled]);
```
ofc it's just an example and `setButtonDisabled` can be moved elsewhere (e.g. to MorseForm), but the general idea is to not keep it in this hook